EIPS/eip-5169.md

```diff
@@ -0,0 +1,191 @@
+
+Each token contract has one `scriptURI`  function to return the download URI to a client script.
+
+Concretely each element in the array contains a URI to download the script itself.
+
+The script provides a client-side executable to the hosting token. Examples of such a script could be:
+
+- A 'miniDapp', which is a cut-down dapp tailored for a single token.
+- A 'TokenScript' which provides [T.I.P.S.](https://tokenscript.org/TIPS.html) from a browser wallet.
+- An extension that is downloadable to the hardware wallet with an extension framework, such as Ledger.
+
+#### Script location
+
+While the most straightforward solution to facilitate specific script usage associated with NFTs, is clearly to store such a script on the smart contract. However, this has several disadvantages: 
+
+1. The smart contract signing key is needed to make updates, causing the key to become more exposed, as it is used more often. 
+
+2. Updates require smart contract interaction. If frequent updates are needed, smart contract calls can become an expensive hurdle.
+
+3. Storage fee. If the script is large, updates to the script will be costly. A client script is typically much larger than a smart contract.
+
+For these reasons, storing volatile data, such as token enhancing functionality, on an external resource makes sense. Such an external resource can be either be  hosted centrally, such as through a cloud provider, or privately hosted through a private server, or decentralized hosted, such as the interplanetary filesystem.
+
+While centralized storage for a decentralized functionality goes against the ethos of web3, fully decentralized solutions may come with speed, price or space penalties. This ERC handles this by allowing the function `ScriptURI` to return multiple URIs, which could be a mix of centralized, individually hosted and decentralized locations.
+
+While this ERC does not dictate the format of the stored script, the script itself could contain pointers to multiple other scripts and data sources, allowing for advanced ways to expand token scripts, such as lazy loading. 
+The handling of the integrity of such secondary data sources is left dependent on the format of the script. For example, HTML format uses [the `integrity` property](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity), while [signed XML format has `<Manifest/>`](https://www.w3.org/TR/xmldsig-core2/#sec-Manifest).
+
+#### Overview
+
+With the discussion above in mind, we outline the solution proposed by this ERC. For this purpose, we consider the following variables:
+
+- `SCPrivKey`: The private signing key to administrate a smart contract implementing this ERC. Note that this doesn't have to be a new key especially added for this ERC. Most smart contracts made today already have an administration key to manage the tokens issued. It can be used to update the `scriptURI`.
```

Comment:
```suggestion
- `SCPrivKey`: The private signing key to administrate a smart contract implementing this EIP. Note that this doesn't have to be a new key especially added for this EIP. Most smart contracts made today already have an administration key to manage the tokens issued. It can be used to update the `scriptURI`.
```